tests/unit/expiry/ExpiryDateStrategyFactory.unitTest.ts

```diff
@@ -40,71 +47,59 @@ describe("ExpiryDateStrategyFactory", () => {
             invalidConfig
           );
         } catch (error) {
-          expect.assertions(1);
           expect(error).toEqual(new Error(ERRORS.ExpiryConfigMissing));
         }
       });
     });
 
     describe("when valid test type id is passed and more than one strategy is returned", () => {
-      it("should throw error", () => {
+      it("should throw an appropriate error", () => {
+        expect.assertions(1);
         try {
+          //  @ts-ignore
+          jest.spyOn(ExpiryDateStrategyFactory, 'getStrategyMapping').mockReturnValue(duplicatedConfig.psv);
+
           const testTypeForExpiry: TestTypeForExpiry = {
             testType: { testTypeId: "3" } as TestType,
             vehicleType: VEHICLE_TYPE.PSV,
             recentExpiry: new Date(0),
             hasHistory: true,
-            hasRegistration: false
+            hasRegistration: true
           };
           ExpiryDateStrategyFactory.GetExpiryStrategy(
             testTypeForExpiry,
             new DateProvider()
           );
         } catch (error) {
-          expect.assertions(1);
           expect(error).toEqual(new Error("Multiple strategies found!"));
         }
       });
     });
-  });
-  context("for valid strategy mapping", () => {
-    describe("when wrong test type id is passed", () => {
-      it("should return error on getExpiryDate", () => {
-        try {
-          const testTypeForExpiry: TestTypeForExpiry = {
-            testType: { testTypeId: "100" } as TestType,
-            vehicleType: VEHICLE_TYPE.HGV,
-            recentExpiry: new Date(0),
-            hasHistory: true,
-            hasRegistration: false
-          };
-          ExpiryDateStrategyFactory.GetExpiryStrategy(
-            testTypeForExpiry,
-            new DateProvider()
-          ).getExpiryDate();
-        } catch (error) {
-          expect.assertions(1);
-          expect(error).toEqual(new Error(ERRORS.MethodNotImplemented));
-        }
-      });
-    });
 
-    describe("when correct test type id is passed", () => {
-      it("should return correct strategy", () => {
+    describe("Resolving correct strategies", () => {
+      test.each`
+      hasHistory  |  hasRegistration  | testTypeId  | vehicleType         | expectedStrategy
+      ${true}     |  ${false}         | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvMostRecentExpiryStrategy}
+      ${true}     |  ${true}          | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvMostRecentExpiryStrategy}
+      ${false}    |  ${true}          | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvRegistrationAnniversaryStrategy}
+      ${false}    |  ${false}         | ${'3'}      | ${VEHICLE_TYPE.PSV} | ${PsvDefaultExpiryStrategy}
+      ${true}     |  ${false}         | ${'100'}    | ${VEHICLE_TYPE.HGV} | ${NoImplementationStrategy}
+    `('for testTypeId $testTypeId  with hasHistory $hasHistory and hasRegistration $hasRegistration it should return $expectedStrategy',
```

Comment:
Can we use also $vehicleType in the description? Would it be useful?

Reply:
Done